packager/packager.cc

```diff
@@ -1465,6 +1467,16 @@ vector<ast::ParsedFile> Packager::run(core::GlobalState &gs, WorkerPool &workers
         core::packages::UnfreezePackages packages = gs.unfreezePackages();
         for (auto &file : files) {
             if (FileOps::getFileName(file.file.data(gs).path()) == PACKAGE_FILE_NAME) {
+                if (file.file.data(gs).strictLevel == core::StrictLevel::Ignore) {
+                    // if the `__package.rb` file is at `typed:
+                    // ignore`, then we haven't even parsed it, which
+                    // means none of the other stuff here is going to
+                    // actually work (since it all assumes we've got a
+                    // file to actually analyze.) If we've got a
+                    // `typed: ignore` package, then skip it.
+                    continue;
+                }
```

Comment:
For my own edification, how does this interact with:

https://github.com/sorbet/sorbet/blob/e23d803535bd2f8454d3fc0be79be1725a7bcc0e/packager/packager.cc#L1365-L1370

Is `Ignore` meant to be a temporary measure for certain `__package.rb` files or something else?

Reply:
I don't think we intend to use `typed: ignore` files for packages, but we should handle them gracefully. In general, `typed: ignore` tells Sorbet, "Please don't even look at this file," so I think here it's reasonable to keep that same contract. I can imagine, for example, someone wanting to experiment with, "What if this package wasn't even here?" and therefore changing it to `typed: ignore`.

It's worth noting that the above error _doesn't_ show up if we mark a package file as `typed: ignore`, since `typed: ignore` will silence all errors.